packages/forklift-console-plugin/src/modules/Providers/views/migrate/ProvidersCreateVmMigrationPage.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { FC, useContext, useEffect } from 'react';
+import { useHistory } from 'react-router';
+import { useForkliftTranslation } from 'src/utils/i18n';
+import { useImmerReducer } from 'use-immer';
+
+import { ProviderModelGroupVersionKind, ProviderModelRef, V1beta1Provider } from '@kubev2v/types';
+import { useK8sWatchResource } from '@openshift-console/dynamic-plugin-sdk';
+import { Button, Divider, Flex, FlexItem, PageSection, Title } from '@patternfly/react-core';
+
+import { useToggle } from '../../hooks';
+import { getResourceUrl } from '../../utils';
+
+import { setAvailableProviders } from './actions';
+import { PlansCreateForm } from './PlansCreateForm';
+import { CreateVmMigrationContext } from './ProvidersCreateVmMigrationContext';
+import { createInitialState, reducer } from './reducer';
+
+const ProvidersCreateVmMigrationPage: FC<{
+  namespace: string;
+}> = ({ namespace }) => {
+  const { t } = useForkliftTranslation();
+  const history = useHistory();
+
+  const { data: { selectedIds: selectedVmIds = [], provider: sourceProvider = undefined } = {} } =
+    useContext(CreateVmMigrationContext);
+  // the page was entered directly without choosing the VMs
+  const emptyContext = !selectedVmIds?.length || !sourceProvider;
+  // redirect to provider list if the page is entered directly
+  useEffect(() => {
+    if (emptyContext) {
+      history.push(
+        getResourceUrl({
+          reference: ProviderModelRef,
+          namespace: namespace,
+        }),
+      );
+    }
+  }, [emptyContext]);
+
+  const [state, dispatch] = useImmerReducer(
+    reducer,
+    { namespace, sourceProvider, selectedVmIds },
+    createInitialState,
+  );
+
+  const [providers] = useK8sWatchResource<V1beta1Provider[]>({
+    groupVersionKind: ProviderModelGroupVersionKind,
+    namespaced: true,
+    isList: true,
+    namespace,
+  });
+  useEffect(() => dispatch(setAvailableProviders(providers ?? [])), [providers]);
+
+  const [isLoading, toggleIsLoading] = useToggle();
+  const onUpdate = toggleIsLoading;
+
+  if (emptyContext) {
+    // wait for redirect triggered from useEffect
+    return <></>;
+  }
+
+  return (
+    <PageSection>
+      <Title headingLevel="h2">{t('Create Migration')}</Title>
+
+      <PlansCreateForm state={state} dispatch={dispatch} />
+
+      <Divider className="forklift-section-create-deviser" />
+
+      <Flex>
+        <FlexItem>
+          <Button
+            variant="primary"
```

Comment:
we will have "create and edit" `primary` that will create a plan and send you to the details page
( ?? assuming most users will want to edit the plan before running it ?? )
and "create and start" `secondary` that will create the plan, and a migration objects
and cancel

Reply:
> we will have "create and edit" primary 
>
just to confirm:  this is a new approach, right? in the first version it was named "create and customize" and was a secondary button for advance use cases.
( [this comment ](https://github.com/kubev2v/forklift-console-plugin/issues/801#issuecomment-1883156188))

Reply:
> it was named "create and customize" and was a secondary button for advance use cases.

**a**. yes
**b**. customization will be done in the "details" pages, we will have tab for "hooks" where users will edit the hook, tab for "mappings" where users will be able to edit "mappings" , tab for vms ( maybe not editable ? maybe reuse the table we use in the providers and create ?) ...
**c**. since we are **very** short on time, in case we will not have time to create the new details page, a fall back will be customize/edit in the old wizard

Reply:
to be clear the string now is "create and edit" ( since the planed behavior is going to the "edit" page )

p.s.
text are easy to change, we can try out other strings if we think we can do better